enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -288,25 +288,31 @@ impl<'a> Handler<'a> {
         let buf = self.aex.gpr.rsi.raw() as *const u8;
         let size = self.aex.gpr.rdx.raw();
 
-        // Allocate some unencrypted memory.
-        let map = match self.ualloc(size) {
-            Err(errno) => return errno as u64,
-            Ok(map) => map,
-        };
+        // Copy the encrypted input into unencrypted memory.
+        let input = unsafe { from_raw_parts(buf, size as usize) };
+        let ubuf = unsafe { from_raw_parts_mut(&self.block.buf as *const _ as _, size as usize) };
```

Comment:
I think what you want is something like this:

```suggestion
        let input: &[u8] = unsafe { from_raw_parts(self.aex.gpr.rsi.into(), self.aex.gpr.rdx.into()) };
        
        if input.len() > self.block.buf.len() { return -EMSGSIZE; }
        self.block.buf[..input.len()].copy_from_slice(input);
};
```

Reply:
I used this except I kept `size` as a variable since it's used several times.